src/main/java/org/example/c17inputoutput/ReadingAFileChallenge.java

```diff
@@ -0,0 +1,85 @@
+package org.example.c17inputoutput;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class ReadingAFileChallenge {
+    public static void main(String[] args) {
+
+        try (BufferedReader br = new BufferedReader(
+                new FileReader("testfile.txt"))) {
+
+//            System.out.printf(" %,d lines in file%n", br.lines().count());
+
+            Pattern pattern = Pattern.compile("\\p{javaWhitespace}+");
+//            System.out.printf("%,d words in file%n",
+//                    br.lines()
+////                            .flatMap(pattern::splitAsStream)
+//                            .flatMap(l -> Arrays.stream(l.split(pattern.toString())))
+//                            .count());
+
+//            System.out.printf("%,d words in file%n",
+//                    br.lines()
+//                            .mapToLong(l -> l.split(pattern.toString()).length)
+//                            .sum());
+
+            List<String> excluded = List.of(
+                    "grand",
+                    "canyon",
+                    "retrieved",
+                    "archived",
+                    "service",
+                    "original");
+            var result = br.lines()
+                    .flatMap(pattern::splitAsStream)
+                    .map(w -> w.replaceAll("\\p{Punct}", ""))
+                    .filter(w -> w.length() > 4)
+                    .map(String::toLowerCase)
+                    .filter(w -> !excluded.contains(w))
+                    .collect(Collectors.groupingBy(w -> w, Collectors.counting()));
+
+            result.entrySet().stream()
+                    .sorted(Comparator.comparing(Map.Entry::getValue,
+                            Comparator.reverseOrder()))
+                    .limit(10)
+                    .forEach(e -> System.out.println(
+                            e.getKey() + " - " + e.getValue() + " times"));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        System.out.println("--------------------");
+        String input = null;
+        try {
+            input = Files.readString(Path.of("testfile.txt"));
+            input = input.replaceAll("\\p{Punct}", "");
+
+            Pattern pattern = Pattern.compile("\\w{5,}");
```

Comment:
Give this Pattern object a suggestive name.

Reply:
Done. New name splitInWordsWithMoreThan5Char

Reply:
Same comment as above.

Reply:
moreThan5Char

Reply:
`\w`means any word character(letter, number, underscore). Char means any character which includes more than word characters.